### ISSUE TITLE
Inconsistent configuration settings for `start` and `import` commands

### ISSUE THREAD

**Issue by andreaTP** [keycloak/keycloak#11767]
### Describe the bug

When starting a pre-augmented Keycloak instance configuration is picked up differently in between the `start` and the `import` command.

Reproducer:

- Build a container `test-custom-keycloak` from a `Dockerfile` as follows ( `docker build . -t test-custom-keycloak` ):
```Dockerfile
FROM quay.io/keycloak/keycloak:18.0.0
RUN /opt/keycloak/bin/kc.sh build --db=postgres --health-enabled=true
```
- the `start` commands:
```bash
docker run --rm -e KC_DB=postgres -e KC_DB_URL_HOST=foo test-custom-keycloak start --hostname-strict=false
```
and
```bash
docker run --rm -e KC_DB_URL_HOST=foo test-custom-keycloak start --hostname-strict=false
```
returns the expected result:
```
java.net.UnknownHostException: foo
```
- instead the `import` command:
```bash
docker run --rm -e KC_DB=postgres -e KC_DB_URL_HOST=foo test-custom-keycloak import --dir=/tmp
```
returns the expected result:
```
java.net.UnknownHostException: foo
```
but the command:
```bash
docker run --rm -e KC_DB_URL_HOST=foo test-custom-keycloak import --dir=/tmp
```
fails with:
```
Connection to localhost:5432 refused.
```
meaning that the option `KC_DB_URL_HOST` has been ignored.

The expected result is that the last command fails to try to resolve `foo` as the rest.

### Version

18.0.0

**Comment by DGuhr** [comment]
This one should also be fixed when the fix from #11902 makes it.

**Comment by pedroigor** [comment]
Closed by https://github.com/keycloak/keycloak/pull/12157.
